Objects/unicodeobject.c

```diff
@@ -118,14 +118,14 @@ NOTE: In the interpreter's initialization phase, some globals are currently
     (assert(_PyUnicode_CHECK(op)),                      \
      PyUnicode_IS_COMPACT_ASCII(op) ?                   \
          ((char*)(_PyASCIIObject_CAST(op) + 1)) :       \
-         _PyUnicode_UTF8(op))
+        FT_ATOMIC_LOAD_PTR(_PyUnicode_UTF8(op)))
 #define _PyUnicode_UTF8_LENGTH(op)                      \
     (_PyCompactUnicodeObject_CAST(op)->utf8_length)
 #define PyUnicode_UTF8_LENGTH(op)                       \
     (assert(_PyUnicode_CHECK(op)),                      \
      PyUnicode_IS_COMPACT_ASCII(op) ?                   \
          _PyASCIIObject_CAST(op)->length :              \
-         _PyUnicode_UTF8_LENGTH(op))
+         FT_ATOMIC_LOAD_SSIZE_RELAXED(_PyUnicode_UTF8_LENGTH(op)))
```

Comment:
No atomic operation here.

Reply:
I think if we need an atomic operation here to silence some TSAN warning, it's a sign of a bug somewhere else.

The atomic synchronization should happen on just one field: `utf8`. If `utf8` is not-NULL, then `utf8_length` should be valid (and not require any atomics to read).